sqlglot/dialects/mysql.py

```diff
@@ -555,11 +555,22 @@ class Generator(generator.Generator):
             exp.WeekOfYear: rename_func("WEEKOFYEAR"),
         }
 
-        TYPE_MAPPING = generator.Generator.TYPE_MAPPING.copy()
-        TYPE_MAPPING.pop(exp.DataType.Type.MEDIUMTEXT)
-        TYPE_MAPPING.pop(exp.DataType.Type.LONGTEXT)
-        TYPE_MAPPING.pop(exp.DataType.Type.MEDIUMBLOB)
-        TYPE_MAPPING.pop(exp.DataType.Type.LONGBLOB)
+        TYPE_MAPPING = {
+            **{
+                k: v
+                for k, v in generator.Generator.TYPE_MAPPING.items()
+                if k
+                not in {
+                    exp.DataType.Type.MEDIUMTEXT,
+                    exp.DataType.Type.LONGTEXT,
+                    exp.DataType.Type.MEDIUMBLOB,
+                    exp.DataType.Type.LONGBLOB,
+                }
+            },
```

Comment:
This is going to run for every key in `TYPE_MAPPING`, right? Not sure why this change was necessary.

Reply:
just do 
```suggestion
TYPE_MAPPING = TYPE_MAPPING.copy()

for from, to in (
  (MEDIUMTEXT, None),
  (TIMESTAMP, "DATETIME")
 : 
 if to:
   TYPE_MAPPING[from] = to
   els:
   TYPE_MAPPING.pop(from)

```

Reply:
Replaced with approach used in previous PR